sgkit_vcf/vcf_partition.py

```diff
@@ -0,0 +1,142 @@
+from pathlib import Path
+from typing import Any, Optional, Sequence
+
+import numpy as np
+from cyvcf2 import VCF
+
+from sgkit.typing import PathType
+from sgkit_vcf.csi import read_csi
+from sgkit_vcf.tbi import read_tabix
+from sgkit_vcf.utils import ceildiv, get_file_length
+
+
+def region_string(contig: str, start: int, end: Optional[int] = None) -> str:
+    if end is not None:
+        return f"{contig}:{start}-{end}"
+    else:
+        return f"{contig}:{start}-"
+
+
+def get_tabix_path(vcf_path: PathType) -> Optional[Path]:
+    tbi_path = Path(vcf_path).parent / (Path(vcf_path).name + ".tbi")
+    if tbi_path.exists():
+        return tbi_path
+    else:
+        return None
+
+
+def get_csi_path(vcf_path: PathType) -> Optional[Path]:
+    csi_path = Path(vcf_path).parent / (Path(vcf_path).name + ".csi")
+    if csi_path.exists():
+        return csi_path
+    else:
+        return None
+
+
+def read_index(index_path: Path) -> Any:
+    if index_path.suffix == ".tbi":
+        return read_tabix(index_path)
+    elif index_path.suffix == ".csi":
+        return read_csi(index_path)
+    else:
+        raise ValueError("Only .tbi or .csi indexes are supported.")
+
+
+def get_sequence_names(vcf_path: Path, index: Any) -> Any:
+    try:
+        return index.sequence_names
```

Comment:
A comment would be helpful here - presumably one of the index types doesn't have a ``sequence_names`` attribute?

Reply:
That's right tbi stores sequence names, csi doesn't. Added a comment.

https://github.com/pystatgen/sgkit-vcf/pull/1/commits/e01e2482b7dbb651bf6a694c985cb6a8d62448fc